src/models/components/reward/curiosity_reward.py

```diff
@@ -0,0 +1,13 @@
+import torch.nn as nn
+from torch import Tensor
+
+from .reward import Reward
+
+
+class CuriosityReward(Reward):
+    def __init__(self):
+        super().__init__()
+        self.mseloss = nn.MSELoss(reduction="none")
+
+    def reward(self, obs_hat: Tensor, obs: Tensor) -> Tensor:
+        return self.mseloss(obs_hat, obs).mean(dim=-1)
```

Comment:
スカラーで返すのでdimの指定はいらないと思います。
```suggestion
        return self.mseloss(obs_hat, obs).mean()
```